library/sha256.c

```diff
@@ -143,6 +243,132 @@ static const uint32_t K[] =
     0x90BEFFFA, 0xA4506CEB, 0xBEF9A3F7, 0xC67178F2,
 };
 
+#endif
+
+#if defined(MBEDTLS_SHA256_USE_A64_CRYPTO_IF_PRESENT) || \
+    defined(MBEDTLS_SHA256_USE_A64_CRYPTO_ONLY)
+
+#if defined(MBEDTLS_SHA256_USE_A64_CRYPTO_ONLY)
+#  define mbedtls_internal_sha256_process_many_a64_crypto mbedtls_internal_sha256_process_many
```

Comment:
Just a common question, what does the word "many" mean?

Reply:
The old `mbedtls_internal_sha256_process()` would process one block at a time. This one processes many (>= 1) blocks at a time. I didn't want to make the name even longer (e.g. `mbedtls_internal_sha256_process_multiple_blocks_a64_crypto()`) and `_many` is a convention already used in Mbed TLS - there are already `ecp_normalize_jac_many()` and `test_mpi_random_many()`

Reply:
Will it support the multi-buffer processing?  If no multi-buffer, did we break up the dependency of blocks to make the parallel processing smoothly?  I just remember any input of blcok(i) will depend on the output of last block(i - 1),  seems there are data dependency.

Reply:
There is no parallel multi-buffer processing.

Yes, each block depends on the previous one - the difference between the old (process one block at a time) and the new (process many blocks at a time) is

OLD:

```
mbedtls_sha256_update(...)
{
    :
    while (data_len >= BLOCK_LEN)
    {
        process_one_block(data)
        data += BLOCK_LEN;
        data_len -= BLOCK_LEN;
    }
    :
}

int process_one_block(const uint8_t data[BLOCK_LEN])
{
    ...
}
```

NEW:

```
mbedtls_sha256_update(...)
{
    :
    if (data_len >= BLOCK_LEN)
    {
        size_t processed = process_one_block(data)
        data += processed;
        data_len -= processed;
    }
    :
}

size_t process_many_blocks(const uint8_t *data, size_t data_len)
{
    ...
}
```

This specific change got nearly 10% performance improvement